### DISCUSSION TITLE
Consider exporting a `Signal` type

### DISCUSSION THREAD

**Discussion by jorroll** [solidjs/solid#446] [Ideas]
As the name suggests, this is a feature request to define and export `export type Signal<T> = () => T` (I'm not actually set on a specific name for this type). While this is a very simple type alias, I think using this type alias throughout the codebase would make reading `solid-js` code and jsDocs generated by vscode easier. At the moment, when looking at a type declaration, sometimes I need to remind myself that the `someFunction<T>(value: () => T)` declaration is expecting a `solid-js` signal (`someFunction<T>(value: Signal<T>)` would be more intuitive I think).

**Comment by ryansolid** [comment]
Sounds good. As long as it wouldn't encourage people to use signals `createSignal` where they could just use functions. I'm still learning TS best practices 2 years later.

**Reply by jorroll** [reply]
> As long as it wouldn't encourage people to use signals `createSignal` where they could just use functions.

Ya, I understand this concern and I honestly don't know. I also must admit that I don't think I yet fully understand how signals compare with simple functions in Solid. I think I saw in another discussion that regular functions can be inserted inside JSX and sometimes will produce the same effect as a signal. I haven't yet explored this though.

**Comment by rturnq** [comment]
I think `Signal` would be confusing as it would imply you need to use the signal primitive and I can't think of a scenario where one would want or need to restrict like that.

I can see the benefit of having a type alias which represents a value getter. Having arguments and return types which are functions can be a little confusing to look at so this would help sort of flatten those cases. I'm not sure what a good name is though: `Value`, `Getter`, `Reader`, `Source` some combination?
